### PR TITLE
Core/Scripts: Meathook and Salramm should patrol once they spawn.

### PR DESCRIPTION
**Pull request by FALL1N1**
[CULLING OF STRATHOLME 5MAN DUNGEON] Currently both bosses just spawn and doesn't move at all. Instead they should spawn and start patrolling around.

Note: Waypoints are not blizzlike, i just ran around and generated some random. ss: http://prntscr.com/jyl1lo

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
> Waypoints are not blizzlike, i just ran around and generated some random

better than none!

**Comment by Knindzagxg** [comment]
If i remember, only Meathook roam whole distance u draw, second one only like 30-40 yards of hes spawn location. Will check on retail once i can.

**Comment by FrancescoBorzi** [comment]
Looking for @FALL1N1  or someone else to complete the work that has been previously started here.

**Comment by Si1ker** [comment]
@FrancescoBorzi Hello, this PR got my attention for a bit.

I included the changes proposed here but it didn't compile completely for me.

Besides, I noticed that there is no source about the Meathook and Salramm patrol behavior in this PR, I spent almost an hour trying to find videos on YT, comments on wowhead and wikis about this movement but I only got info about this patrolling behavior on cataclysm and the other expansions ahead but not for woltk, is there any source that can confirm this patrol behavior for those 2 bosses in woltk? Thanks.

**Comment by FrancescoBorzi** [comment]
@Si1ker not that I'm aware of :(

**Comment by BarbzYHOOL** [comment]
Post your errors Silker

**Comment by FALL1N1** [comment]
Paste me the build log and I'll fix it, sadly I cannot build it as I cannot compile on my laptop because it will melt down, got no access to my workstation.

**Comment by BarbzYHOOL** [comment]
@Si1ker

**Comment by Si1ker** [comment]
@BarbzYHOOL thx for the tag 👍 
@FALL1N1 After including the changes proposed, that's what I've got:

1>------ Build started: Project: revision.h, Configuration: Release x64 ------
2>------ Skipped Build: Project: clean_cotire, Configuration: Release x64 ------
2>Project not selected to build for this solution configuration 
1>--
1>    Could not find a proper repository signature (hash) - you may need to pull tags with git fetch -t
1>    Continuing anyway - note that the versionstring will be set to "unknown 1970-01-01 00:00:00 (Archived)"
3>------ Build started: Project: scripts, Configuration: Release x64 ------
3>boss_meathook.cpp
3>boss_salramm.cpp
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_salramm.cpp(51): error C2259: 'boss_salramm::boss_salrammAI': cannot instantiate abstract class
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_salramm.cpp(51): note: due to following members:
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_salramm.cpp(51): note: 'void npc_escortAI::WaypointReached(uint32)': is abstract
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_meathook.cpp(41): error C2259: 'boss_meathook::boss_meathookAI': cannot instantiate abstract class
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\game\AI\ScriptedAI\ScriptedEscortAI.h(77): note: see declaration of 'npc_escortAI::WaypointReached'
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_meathook.cpp(41): note: due to following members:
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\scripts\Kalimdor\CavernsOfTime\CullingOfStratholme\boss_meathook.cpp(41): note: 'void npc_escortAI::WaypointReached(uint32)': is abstract
3>C:\Users\Silker\Desktop\server\azerothcore-wotlk-master\src\server\game\AI\ScriptedAI\ScriptedEscortAI.h(77): note: see declaration of 'npc_escortAI::WaypointReached'
3>Done building project "scripts.vcxproj" -- FAILED.
4>------ Skipped Build: Project: INSTALL, Configuration: Release x64 ------
4>Project not selected to build for this solution configuration 
========== Build: 1 succeeded, 1 failed, 20 up-to-date, 2 skipped ==========

**Comment by FALL1N1** [comment]
thanks, now all we need is a volunteer to test if it works

**Comment by BarbzYHOOL** [comment]
@FALL1N1 thank you!
@Si1ker we count on you (more and more :P)!

**Comment by FALL1N1** [comment]
build on VS19 works just fine, travis is being a bitch.. so in short:
- it works as intended and builds fine
- it does not pass the travis-ci build due to the code being ugly/non-standart

if anyone is interested in fixing the travis - feel free to do so

**Comment by Si1ker** [comment]
@BarbzYHOOL Ok n.n
@FrancescoBorzi I've checked in Warmane, both are walking the same path provided in this PR, ty
@FALL1N1 Now it is building, it compiled with no problem and both bosses follow the WPs in-game, but there's a problem with the map.

When meathook, goes from wp#1 to wp#2, in this transition he goes under map and then emerges in wp#3 to follow towards the wp#4 flying above the map.

2020-03-07 14_48_06-Window
2020-03-07 14_48_21-Window
2020-03-07 14_48_41-Window

Then all you have is a Meathook flying..

In the other hand, we have Salram under map all his way in the same point Meathook had his problem, in the transition from wp#1 to wp#2

2020-03-07 14_58_12-Window


-- I tried adding multiple WPs to have a "better" path but I think it is more related with the vmap..  : /

The problem always starts here: in the smoke at the right side, he goes under map, in the smoke at the left side he emerges and follows the other WPs (meathook flying and Salram under map), at some point for some reason they "update" his movement and goes as expected but again when they reach the [right smoke] the same problem happens.

2020-03-08 22_54_51-Window

2020-03-08 23_05_04-Window

I'll be trying with around 20 WPs to have a more efficient and complex way, after that Illl be out of ideas haha 🐱 , but again, I think is vmap/map related.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

It is like the problem with Kel'Thuzaad adds in Phase 1 of Naxxramas, some mobs generated in these gates go under the map:

unknown

unknown (1)

**Comment by FALL1N1** [comment]
The issue is indeed from the vmaps/mmaps, we have discussed it in a previous issue at stranglethorn vale somewhere near a cliff where they went straight through the cliff instead of finding a path to the top.

**Comment by BarbzYHOOL** [comment]
damned mmaps like here https://github.com/azerothcore/azerothcore-wotlk/issues/2688

ok so what must we do with the PR now?

**Comment by FrancescoBorzi** [comment]
General tip for testers: when finding issues while testing, please specify whether they are regressions (new issues that weren't on master) or not. In the second case, the PR could still be merged.

**Comment by BarbzYHOOL** [comment]
if the map issue is fixed, what about this PR then?

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp [hunk 1]
```diff
@@ -103,7 +118,7 @@ class boss_salramm : public CreatureScript
                     }
         }
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp:120]: void WaypointReached(uint32 waypointId) { }

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp [hunk 2]
```diff
@@ -71,7 +87,7 @@ class boss_meathook : public CreatureScript
             Talk(SAY_SLAY);
         }
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp:89]: void WaypointReached(uint32 waypointId) { }

Comment (pklloveyou) [reply src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp:89]: Fix build

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp (hunk 1 above)
Comment (pklloveyou) [reply src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp:120]: Fix build

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp [hunk 3]
```diff
@@ -47,14 +48,28 @@ class boss_salramm : public CreatureScript
 
     CreatureAI* GetAI(Creature* creature) const
     {
-        return new boss_salrammAI (creature);
+        return GetInstanceAI<boss_salrammAI>(creature);
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp:51]: - return GetInstanceAI<boss_salrammAI>(creature);
+return new boss_salrammAI (creature);

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp [hunk 4]
```diff
@@ -37,14 +39,28 @@ class boss_meathook : public CreatureScript
 
     CreatureAI* GetAI(Creature* creature) const
     {
-        return new boss_meathookAI (creature);
+        return GetInstanceAI<boss_meathookAI>(creature);
```
Comment (pklloveyou) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp:42]: -return GetInstanceAI<boss_meathookAI>(creature);
+return new boss_meathookAI (creature);

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp (hunk 3 above)
Comment (pklloveyou) [reply src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_salramm.cpp:51]: Fix build

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp (hunk 4 above)
Comment (pklloveyou) [reply src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/boss_meathook.cpp:42]: Fix build

**Comment by Kitzunu** [comment]
I will close this as abandoned.
Feel free to reopen whenever someone wants to take it over :)

**Pull request closed without merging**